File: TopCoder01/Ruby/KiwiJuiceEasy.rb
class KiwiJuiceEasy
  def thePouring(capacities, bottles, fromId, toId)
    i = 0
    while i < fromId.length
      sum = bottles[fromId[i]] + bottles[toId[i]]
      bottles[toId[i]] = [sum, capacities[toId[i]]].min
      bottles[fromId[i]] = sum - bottles[toId[i]]
      i += 1
    end
    return bottles
  end
 
end

File: TopCoder01/Scala/KiwiJuiceEasy.scala
class KiwiJuiceEasy {
  def thePouring(capacities: Array[Int], bottles: Array[Int],
      fromId: Array[Int], toId: Array[Int]): Array[Int] = {
    for(i <- 0 until fromId.length){
       var sum = bottles(fromId(i)) + bottles(toId(i))
       bottles(toId(i)) = math.min(sum, capacities(toId(i)))
       bottles(fromId(i)) = sum - bottles(toId(i))
    }
    return bottles
  }

}